### PR TITLE
Edit event

### PR DESCRIPTION
**Pull request by chacetrev10**

### CODE REVIEW

File: PurpleNut/urls.py [hunk 1]
```diff
@@ -33,7 +33,9 @@
     url(r'^login/',auth_views.LoginView.as_view(template_name='users/login.html') ,name='login'),
     url(r'^logout/',auth_views.LogoutView.as_view(template_name='users/logout.html') ,name='logout'),
     url(r'^profile/',user_views.profile,name='profile'),
-    url('^profile_edit/', user_views.edit_profile, name = 'edit_profile')
+    url(r'^profile_edit/', user_views.edit_profile, name = 'edit_profile'),
```
Comment (royceschultz) [inline PurpleNut/urls.py]: Merge conflict: Use, r'^profile/(\d+)/ and r'profile/edit/' from the master branch

File: events/templates/create_event.html [hunk 2]
```diff
@@ -2,9 +2,9 @@
 {% load crispy_forms_tags %}
 {% block content %}
 <h2>{{note}}</h2>
-<form action="{%url 'create event'%}" method="post">
+<form action="" method="POST">
   {% csrf_token %}
   {{ form|crispy }}
-  <input type="submit" name="" value="Create">
+  <input type="submit" name="" value="Save">
```
Comment (royceschultz) [inline events/templates/create_event.html]: This form has no action? Is it because it's a model form?

File: events/views.py [hunk 3]
```diff
@@ -36,3 +36,19 @@ def create_event(request):
     else: # If logged in user is accessing the page through a normal GET request
         form = EventForm()
         return render(request, 'create_event.html',{'form':form})
+
+
+def edit_event(request, id):
+    event = Event.objects.get(id=id)
```
Comment (royceschultz) [inline events/views.py]: What if event does not exist? See my implementation of the profiles page in my recent pull request.

File: events/templates/create_event.html (hunk 2 above)
Comment (chacetrev10) [reply events/templates/create_event.html]: Needed so edit could inherit from this from and doesn't create a whole new event. Still functions for creating new events

File: events/views.py [hunk 4]
```diff
@@ -36,3 +36,19 @@ def create_event(request):
...
+    form =EventForm( instance =event)
```
Comment (royceschultz) [inline events/views.py]: We need to verify the user editing the event is the author of the event.

File: PurpleNut/urls.py (hunk 1 above)
Comment (chacetrev10) [reply PurpleNut/urls.py]: made changes

File: events/views.py [hunk 5]
```diff
@@ -36,3 +36,19 @@ def create_event(request):
...
+    if request.method == 'POST':
+        form = EventForm( request.POST, instance =event)
+        if form.is_valid():
+            form.save()
+            messages.success(request,f'Changes have been made to your event')
+            return redirect('home')
```
Comment (royceschultz) [inline events/views.py]: Maybe redirect to the event detail page

File: events/views.py [hunk 6]
```diff
@@ -36,3 +37,24 @@ def create_event(request):
     else: # If logged in user is accessing the page through a normal GET request
         form = EventForm()
         return render(request, 'create_event.html',{'form':form})
+
+
+def edit_event(request, id):
+
+    try:
+        event = Event.objects.get(id=id)
+
```
Comment (royceschultz) [inline events/views.py]: if event.author.id != request.user.id:
    return 404('User does not match Author')

We'll modify later if this is not secure.
If this is the only change, we'll just pull. No need to make a new commit for a small change

File: PurpleNut/urls.py [hunk 7]
```diff
@@ -32,8 +32,14 @@
     url(r'^register/',user_views.register,name='register'),
     url(r'^login/',auth_views.LoginView.as_view(template_name='users/login.html') ,name='login'),
     url(r'^logout/',auth_views.LogoutView.as_view(template_name='users/logout.html') ,name='logout'),
+
+    
+    url(r'^event_edit/(\d+)', event_views.edit_event, name='event edit')
```
Comment (royceschultz) [inline PurpleNut/urls.py]: event/edit/(\d+)
Keep consistent with profile edit

**Pull request merged**